lib/util/ts_parser.js

```diff
@@ -0,0 +1,440 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.util.TsParser');
+
+goog.require('shaka.log');
+goog.require('shaka.util.BufferUtils');
+goog.require('shaka.util.Id3Utils');
+goog.require('shaka.util.Uint8ArrayUtils');
+
+
+/**
+ * @see https://en.wikipedia.org/wiki/MPEG_transport_stream
+ * @export
+ */
+shaka.util.TsParser = class {
+  /** */
+  constructor() {
+    /** @private {?number} */
+    this.pmtId = null;
+
+    /** @private {boolean} */
+    this.pmtParsed = false;
+
+    /** @private {?number} */
+    this.videoPid = null;
+
+    /** @private {?string} */
+    this.videoCodec = null;
+
+    /** @private {!Array.<Uint8Array>} */
+    this.videoData = [];
+
+    /** @private {?number} */
+    this.audioPid = null;
+
+    /** @private {?string} */
+    this.audioCodec = null;
+
+    /** @private {!Array.<Uint8Array>} */
+    this.audioData = [];
+
+    /** @private {?number} */
+    this.id3Pid = null;
+
+    /** @private {!Array.<Uint8Array>} */
+    this.id3Data = [];
+
+    /** @private {?Uint8Array} */
+    this.remainderData = null;
+  }
+
+  /**
+   * Check if the passed data corresponds to an MPEG2-TS
+   * @param {Uint8Array} data
+   * @return {!shaka.util.TsParser}
+   */
+  parse(data) {
+    let unknownPIDs = false;
+
+    let len = data.length;
+    if (this.remainderData) {
+      data = shaka.util.Uint8ArrayUtils.concat(this.remainderData, data);
+      len = data.length;
+      this.remainderData = null;
+    }
+
+    if (len < 188) {
+      this.remainderData = data;
+      return this;
+    }
+
+    const syncOffset = Math.max(0, shaka.util.TsParser.syncOffset(data));
+
+    len -= (len + syncOffset) % 188;
+    if (len < data.byteLength) {
+      this.remainderData = shaka.util.BufferUtils.toUint8(data, len);
+    }
+
+    // loop through TS packets
+    for (let start = syncOffset; start < len; start += 188) {
+      if (data[start] === 0x47) {
+        const stt = !!(data[start + 1] & 0x40);
+        // pid is a 13-bit field starting at the last bit of TS[1]
```

Comment:
It appears to be the last _5_ bits of TS[1].

Reply:
Fixed!